shell/src/main/java/alluxio/cli/bundler/command/CollectLogCommand.java

```diff
@@ -53,15 +157,144 @@ public boolean hasSubCommand() {
+      System.out.format("Include the following filename prefixes: %s%n", toInclude);
+      mIncludedPrefix.addAll(toInclude);
+    }
+    if (cl.hasOption(EXCLUDE_OPTION_NAME)) {
+      mExcludedPrefix = parseFileNames(cl.getOptionValue(EXCLUDE_OPTION_NAME));
+      System.out.format("Exclude the following filename prefixes: %s%n", mExcludedPrefix);
+    }
+
+    // Check file timestamps
+    boolean checkTimeStamp = false;
+    if (cl.hasOption(START_OPTION_NAME)) {
+      String startTimeStr = cl.getOptionValue(START_OPTION_NAME);
+      mStartTime = parseDateTime(startTimeStr);
+      System.out.format("Time window start: %s%n", mStartTime);
+      checkTimeStamp = true;
+    }
+    if (cl.hasOption(END_OPTION_NAME)) {
+      String endTimeStr = cl.getOptionValue(END_OPTION_NAME);
+      mEndTime = parseDateTime(endTimeStr);
+      System.out.format("Time window end: %s%n", mEndTime);
+      checkTimeStamp = true;
+    }
+    if (mStartTime != null && mEndTime != null && mStartTime.isAfter(mEndTime)) {
+      System.err.format("ERROR: Start time %s is later than end time %s!%n",
+              mStartTime, mEndTime);
+    }
+
+    if (!mLogDir.exists()) {
+      System.err.format("ERROR: Alluxio log directory %s does not exist!%n", mLogDirPath);
+      return -1;
+    }
+
+    List<File> allFiles = CommonUtils.recursiveListLocalDir(mLogDir);
+    for (File f : allFiles) {
+      String relativePath = getRelativePathToLogDir(f);
+      try {
+        if (!shouldCopy(f, relativePath, checkTimeStamp)) {
+          continue;
+        }
+        File targetFile = new File(mWorkingDirPath, relativePath);
+        FileUtils.copyFile(f, targetFile, true);
+      } catch (IOException e) {
+        System.err.format("ERROR: file %s not found %s%n", f.getCanonicalPath(), e.getMessage());
+      }
+    }
 
     return 0;
   }
 
+  private String getRelativePathToLogDir(File f) {
+    return mLogDirUri.relativize(f.toURI()).getPath();
+  }
+
+  private boolean shouldCopy(File f, String relativePath, boolean checkTimeStamp)
+          throws IOException {
+    if (!fileNameIsWanted(relativePath)) {
+      return false;
+    }
+    if (checkTimeStamp) {
+      if (!fileTimeStampIsWanted(f)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean fileNameIsWanted(String fileName) {
+    if (mExcludedPrefix != null) {
+      for (String x : mExcludedPrefix) {
+        if (fileName.startsWith(x)) {
+          return false;
+        }
+      }
+    }
+    for (String s : mIncludedPrefix) {
+      if (fileName.startsWith(s)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean fileTimeStampIsWanted(File f) throws IOException {
+    long timestamp = f.lastModified();
+    LocalDateTime fileEndTime =
+            LocalDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneId.systemDefault());
+
+    // Infer file start time by parsing the first bunch of rows
```

Comment:
We should guard this with `if (mStartTime != null)` and in the block, return false if it fails.

Reply:
same